src/views/CharacterDetails.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { fetchDetails } from '../services/fetch.js';
+
+export default function CharacterDetails() {
+  const [character, setCharacter] = useState({});
+  const { id } = useParams();
+
+  useEffect(() => {
+    const getDetails = async () => {
+      const data = await fetchDetails(id);
+
+      setCharacter(data);
+    };
+
+    getDetails();
+  }, []);
```

Comment:
Consider adding id to the dependency array here, as we want the useEffect to rerun when the id changes :)